scripts/west_commands/runners/probe_rs.py

```diff
@@ -88,7 +92,12 @@ def do_flash(self, **kwargs):
         download_args = []
         if self.erase:
             download_args += ['--chip-erase']
-        download_args += [self.elf_name]
+
+        if self.hex_file is not None:
+            download_args += ['--binary-format', 'hex']
+            download_args += [self.hex_file]
+        else:
+            download_args += [self.elf_name]
```

Comment:
If nothing is specified, probe-rs uses the preferred format of the target, this might happen to be elf. Let's specify elf here explicitly to be on the safe side.

```suggestion
            download_args += ['--binary-format', 'elf']
            download_args += [self.elf_name]
```